jupyter_client/kernelspec.py

```diff
@@ -161,6 +154,26 @@ def get_kernel_spec(self, kernel_name):
 
         return self.kernel_spec_class.from_resource_dir(resource_dir)
 
+    def get_kernel_spec(self, kernel_name):
+        """Returns a :class:`KernelSpec` instance for the given kernel_name.
+
+        Raises :exc:`NoSuchKernel` if the given kernel name is not found.
+        """
+        d = self.find_kernel_specs()
+        try:
+            resource_dir = d[kernel_name.lower()]
+        except KeyError:
+            raise NoSuchKernel(kernel_name)
+
+        return self._get_kernel_spec_by_name(kernel_name, resource_dir)
+
+    def find_all_specs(self):
```

Comment:
let's call this `get_all_specs`. That way `get_` returns kernelspec objects, while `find_` returns locations on disk.

And don't forget to add a docstring.